python/tk_framework_desktopserver/shotgun/api_v2.py

```diff
@@ -829,6 +889,29 @@ def _get_subprocess_kwargs(self):
 
         return kwargs
 
+    def _get_task_parent_entity_type(self, task_id):
+        """
+        Gets the Task entity's parent entity type.
+        """
+        cache = self.WSS_KEY_CACHE[self._wss_key]
+
+        if "task_parent_types" in cache and task_id in cache["task_parent_types"]:
+            logger.debug("Parent entity type found in cache for Task %s.", task_id)
+        else:
+            context = sgtk.context.from_entity(
+                self._engine.sgtk,
+                "Task",
+                task_id,
+            )
+            entity_type = context.entity["type"]
+
+            if "task_parent_types" in cache:
```

Comment:
```setdefault``` retrieves the value for `task_parent_types`. If it doesn't exist, then it sets `{}` as the value and returns it.